### ISSUE TITLE
`indent`: Support JSXText indent

### ISSUE THREAD

**Issue by Daydreamer-riri** [eslint-stylistic/eslint-stylistic#378]
### Clear and concise description of the problem

‌‌‌‌‌‌Based on my practice, the `indent` rule can "almost" replace the `jsx-indent` rule and do a better job. Before I made the fix (#368), there was even a conflict between the two. Moreover, `jsx-indent` would lack indentation in multiline ternary operations. So now I prefer to turn off the `jsx-indent` rule and use only the `indent` rule. However, the reason I used the word "almost" is because `indent` does not handle the following situation:

```jsx
<div>
JSXText  // `indent` does not indent for this
  {'Literal in JSXExpressionContainer'} // `indent` will indent for this node
</div>
```

### Suggested solution

‌‌‌‌‌‌‌Support JSXText indentation for the `indent` rule. Considering that the rule already indents almost all JSX-related nodes (and it handles the overlapping parts of JSX nodes with other nodes quite well), the current lack of support for JSXText might be considered a BUG? I believe this should not be considered a breaking change.

### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Contributes

- [X] If this feature request is accepted, I am willing to submit a PR to fix this issue

**Comment by antfu** [comment]
Sounds good, feel free to send the PR.

Would you suggest implementing the feature-parity and then deprecating just-indent in favor of indent?

**Comment by Daydreamer-riri** [comment]
Yes, I suggest it. I can attempt to make `indent` pass all the test cases of `jsx-indent` to achieve feature parity.

**Comment by antfu** [comment]
Sounds nice if we managed to do that. Feel free to PR, thanks!

**Comment by Daydreamer-riri** [comment]
The task of implementing jsx-indent testing is more complex than initially anticipated due to the need to consider indentation rules for non-jsx code as well. I am still working on this task as quickly as possible. (I was a bit busy last week🥹)

**Comment by JstnMcBrd** [comment]
Now that this feature was added in 2.2.0, will the `jsx-indent` rule be deprecated as suggested earlier?

Also, will the same consolidation happen for `indent-binary-ops` and `jsx-indent-props`, or will those continue to be separate rules? For `indent-binary-ops` at least, the configuration settings are identical to `indent`.

**Comment by Daydreamer-riri** [comment]
indent can also replace the functionality of jsx-indent-props. The other I'm not sure about.
